### DISCUSSION TITLE
IKEA E1743 firmware: different up-to-date versions

### DISCUSSION THREAD

**Discussion by bartplessers** [Koenkk/zigbee2mqtt#12295] [General]
Hi,
I noticed that I have 2 different versions of firmware for this device
- 2.2.010
- 2.3.079

![2022-04-27_14-08-31](https://user-images.githubusercontent.com/10470708/165515069-ba3a7d15-b5af-43d6-adac-2aad98ebee27.png)

Although all devices seems to be up to date....

Can anybody explain this?

kind regards,
Bart Plessers

**Comment by Koenkk** [comment]
Could be an old bug which failed to read the new firmware version after update (solved in 1.25.0). You can read the `swBuildId` of the `genBasic` cluster by clicking on the device and going to the dev console.

**Reply by bartplessers** [reply]
Hi @Koenkk ,

Can you ellaborate this a little bit?
![2022-04-29_10-27-32](https://user-images.githubusercontent.com/10470708/165910191-a0dbf12a-3eac-478b-8817-8eaf3ee79c6c.png)

This is what I'm doing now, but nothing happens
  (1) select device
  (2) dev console
  (3) Cluster -> genBasic
  (4) Attribute swBuildId
  (5) Read

but I don't get any feedback
I'm running 1.25.0 now

This is what I see in the (debug)log:

```
info  2022-04-29 10:32:55: MQTT publish: topic 'zigbee2mqtt/bridge/config', payload '{"commit":"unknown","coordinator":{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20220219,"transportrev":2},"type":"zStack3x0"},"log_level":"debug","network":{"channel":11,"extendedPanID":"0xdddddddddddddddd","panID":6758},"permit_join":false,"version":"1.25.0"}'
debug 2022-04-29 10:33:13: Received MQTT message on 'zigbee2mqtt/my/n00/p03/switch/R18/wandschakelaar/1/set' with data '{"read":{"attributes":["swBuildId"],"cluster":"genBasic","options":{}}}'
debug 2022-04-29 10:33:13: Publishing 'set' 'read' to 'my/n00/p03/switch/R18/wandschakelaar'
```
But I don't receive any response from 'my/n00/p03/switch/R18/wandschakelaar'

kind regards,
B

**Reply by bartplessers** [reply]
OOPS, solved. 
It seems that you have to do an action on the device, i.e. pressing the button....

**Comment by bartplessers** [comment]
This brings me to following:


### Device R16
In the GUI:
![2022-04-29_10-48-57](https://user-images.githubusercontent.com/10470708/165913385-440675c2-d7ba-47ac-87d7-c26ce24f7bd3.png)

Read result of 'genBasic': {"swBuildId":"2.3.079"}

but showing stille 2.2.010 on the OTA interface (see first screenshot on this thread)




### Device R18
'zigbee2mqtt/my/n00/p03/switch/R18/wandschakelaar'

No result in GUI (1):
![2022-04-29_10-54-33](https://user-images.githubusercontent.com/10470708/165914259-49bea3f9-1a3e-441d-bca4-c279955e6da4.png)

But in the logs:
`
Received Zigbee message from 'my/n00/p03/switch/R18/wandschakelaar', type 'readResponse', cluster 'genBasic', data '{"swBuildId":"2.2.010"}' from endpoint 1 with groupID 0
`

So, it's on version 2.2.010, but there seems no update available.
However, same E1743 devices on my network have 2.3.079...


Questions
- is there a way to normalize the interface so that's showing the correct versions?
- how to update the devices where there seems not be an update available (but where there is one)

kind regards,
B

**Reply by Koenkk** [reply]
> is there a way to normalize the interface so that's showing the correct versions?

A restart of z2m should do it (note that this was an old bug which has already been fixed when updating devices through z2m 1.25.0)

> how to update the devices where there seems not be an update available (but where there is one)


Can you provide a debug log when checking for an ota update?

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable debug logging.

**Comment by kopfsick** [comment]
@bartplessers , did you ever resolve this? I have the same issue; a device is on 2.2.010 and reports no available update. The docs on the device page on the zigbee2mqtt site says version 2.3.075 is needed to be able to bind to devices and not groups.

**Comment by ffries** [comment]
Same issue here: I am stuck with version 2.2.010 and no upgrade available.

**Comment by antti365** [comment]
I have an even more interesting versions...

![111](https://github.com/Koenkk/zigbee2mqtt/assets/6865425/45890619-436f-4446-8175-9458b3ba809e)

And by the way, just recently I wrote about another problem with OTA updates for Ikea - https://github.com/Koenkk/zigbee2mqtt/discussions/20437#discussion-6011788

**Comment by hollosipeti** [comment]
I also have an E1743 switch that I cannot upgrade. And unfortunately it eats batteries. Is there no hardware way to update it?
